werkzeug/datastructures.py

```diff
@@ -765,6 +759,8 @@ def __eq__(self, other):
                 return False
         return True
 
+    __hash__ = None
```

Comment:
i recall its a misfeature with eq vs hash - as far as i understood if you specify eq, hash wont be inherited from object
and to properly behave one needs to set it up